marlowe-dashboard-client/src/Dashboard/View.purs

```diff
@@ -32,74 +34,131 @@ import Tooltip.Types (ReferenceId(..))
+    card = view _card state
 
-    cards = view _cards state
+    cardOpen = view _cardOpen state
+
+    mSelectedContractState = preview _selectedContract state
   in
     div
-      [ classNames $ [ "grid", "h-full", "grid-rows-main" ] <> applyWhen menuOpen [ "bg-black" ] ]
-      [ renderHeader walletNickname menuOpen
-      , main
-          [ classNames [ "relative", "px-4", "md:px-5pc" ] ]
-          [ renderMobileMenu menuOpen
-          , div_ $ renderCard currentSlot state <$> cards
-          , renderScreen currentSlot state
+      [ classNames
+          $ [ "h-full", "grid", "grid-rows-dashboard-outer", "transition-all", "duration-500" ]
+          <> applyWhen cardOpen [ "lg:mr-sidebar" ]
+      ]
+      [ dashboardHeader walletNickname menuOpen
+      , div [ classNames [ "relative" ] ] -- this wrapper is relative because the mobile menu is absolutely positioned inside it
+          [ mobileMenu menuOpen
+          , div [ classNames [ "h-full", "grid", "grid-rows-dashboard-inner" ] ]
+              [ dashboardBreadcrumb mSelectedContractState
+              , main
+                  [ classNames $ [ "w-full" ] <> if isJust mSelectedContractState then [] else Css.container ] case mSelectedContractState of
+                  Just contractState -> [ ContractAction <$> contractDetailsCard currentSlot contractState ]
+                  Nothing -> [ contractsScreen currentSlot state ]
+              ]
           ]
-      , renderFooter
+      , dashboardFooter
       ]
 
+dashboardCard :: forall p. Slot -> State -> HTML p Action
+dashboardCard currentSlot state = case view _card state of
+  Just card ->
+    let
+      cardOpen = view _cardOpen state
+
+      walletLibrary = view _walletLibrary state
+
+      currentWalletDetails = view _walletDetails state
+
+      assets = view _assets currentWalletDetails
+
+      walletNicknameInput = view _walletNicknameInput state
+
+      walletIdInput = view _walletIdInput state
+
+      remoteWalletInfo = view _remoteWalletInfo state
+
+      mSelectedContractState = preview _selectedContract state
+
+      templateState = view _templateState state
+    in
+      div
+        [ classNames $ Css.sidebarCardOverlay cardOpen ]
+        [ div
+            [ classNames $ Css.sidebarCard cardOpen ]
+            $ [ a
+                  [ classNames [ "absolute", "top-4", "right-4" ]
+                  , onClick_ CloseCard
+                  ]
+                  [ icon_ Close ]
+              ]
+            <> case card of
+                SaveWalletCard mTokenName -> [ saveWalletCard walletLibrary walletNicknameInput walletIdInput remoteWalletInfo mTokenName ]
+                ViewWalletCard walletDetails -> [ walletDetailsCard walletDetails ]
+                PutdownWalletCard -> [ putdownWalletCard currentWalletDetails ]
+                WalletLibraryCard -> [ walletLibraryScreen walletLibrary ]
+                TemplateLibraryCard -> [ TemplateAction <$> templateLibraryCard ]
```

Comment:
It's better to start using￼ the renderSubmodule version￼￼￼￼￼, as it allows you to add normal HTML and also subcomponents￼￼￼

Reply:
:+1: